pkg/component/observability/monitoring/prometheus/component.go

```diff
@@ -190,12 +200,20 @@ func (p *prometheus) Deploy(ctx context.Context) error {
 		}
 	}
 
+	var cortexConfigMap *corev1.ConfigMap
+	if p.values.Cortex != nil {
+		cortexConfigMap = p.cortexConfigMap()
+		if err := registry.Add(cortexConfigMap); err != nil {
+			return err
+		}
```

Comment:
This could be replaced by just adding `cortexConfigMap` to the registry in the `AddAllAndSerialize(...)` call as it handles `nil` correctly.
Any particular reason why you deviate from this pattern?